src/libertem/io/dataset/memory.py

```diff
@@ -246,14 +260,15 @@ def need_decode(self, read_dtype, roi):
             return True
         return super().need_decode(read_dtype, roi)
 
-    def get_tiles(self, *args, **kwargs):
+    def get_tiles(self, roi=None, *args, **kwargs):
         # force our own tiling_scheme, if a tileshape is given:
         if self._tileshape is not None and self._force_tileshape:
             tiling_scheme = TilingScheme.make_for_shape(
                 tileshape=self._tileshape,
                 dataset_shape=self.meta.shape,
             )
             kwargs.update({"tiling_scheme": tiling_scheme})
+        read_ranges = self._get_read_ranges(tiling_scheme, roi)
         tiles = super().get_tiles(*args, **kwargs)
```

Comment:
`roi` needs to be passed down here, I think

Reply:
Yup, it's incomplete. I'm sorry I left the work halfway on `memory`. I'll get back to it later.

Reply:
Okay, no worries!